packetbeat/protos/dns/dns_test.go

```diff
@@ -649,3 +788,266 @@ func assertAddress(t testing.TB, expected common.IpPortTuple, endpoint interface
 	assert.Equal(t, expected.Src_ip.String(), e.Ip)
 	assert.Equal(t, expected.Src_port, e.Port)
 }
+
+// TCP tests
+
+func testTcpTuple() *common.TcpTuple {
+	t := &common.TcpTuple{
+		Ip_length: 4,
+		Src_ip:    net.IPv4(192, 168, 0, 1), Dst_ip: net.IPv4(192, 168, 0, 2),
+		Src_port: ClientPort, Dst_port: ServerPort,
+	}
+	t.ComputeHashebles()
+	return t
+}
+
+// If a TCP gap (lost packets) happen while we're waiting for parts
```

Comment:
Yes. But I'm not sure how a DNS Stream should be dealt with if a gap occurs. Should we consider that a DNS Stream is not decodable if there is a a gap anyway ? Or that it might be decodable at some point in the future ?
For now we consider that if it not decodable right away, we should drop it.